### PR TITLE
[airvisualnode] Add null annotations

### PR DESCRIPTION
**Pull request by lsiepel**
Signed-off-by: Leo Siepel <leosiepel@gmail.com>

added null annotations, some very minor refactoring.
Left one `UnusedPrivateField` warning that allready existed. It is part of json/dto and might be essential in request/response model binding.

Edit: I thik it is ready for a review. Maybe not merge it so short before 3.4.0 as it does not solve bugs (but tries to prevent them) a JAR is available here: https://1drv.ms/u/s!AnMcxmvEeupwjpdU8sz_NrUHNz7Diw?e=DN2WPo

### CODE REVIEW

File: bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java [hunk 1]
```diff
@@ -63,20 +67,23 @@ protected void startScan() {
     @Override
     protected void startBackgroundDiscovery() {
         logger.debug("Starting background discovery");
-        backgroundDiscoveryFuture = scheduler.scheduleWithFixedDelay(this::scan, 0, 5, TimeUnit.MINUTES);
+        ScheduledFuture<?> localDiscoveryFuture = backgroundDiscoveryFuture;
+        if (localDiscoveryFuture == null || localDiscoveryFuture.isCancelled()) {
+            backgroundDiscoveryFuture = scheduler.scheduleWithFixedDelay(this::startScan, 30, REFRESH_MINUTES,
```
Comment (jlaur) [inline bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java:72]: Did you mean to change `initialDelay` from 0 to 30 minutes and `delay` from 5 to 600 minutes?

File: bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java [hunk 2]
```diff
@@ -63,20 +67,23 @@ protected void startScan() {
...
 
-    private void cancelBackgroundDiscoveryFuture() {
-        if (backgroundDiscoveryFuture != null && !backgroundDiscoveryFuture.isDone()) {
-            backgroundDiscoveryFuture.cancel(true);
-            backgroundDiscoveryFuture = null;
+        ScheduledFuture<?> localDiscoveryFuture = backgroundDiscoveryFuture;
+        if (localDiscoveryFuture != null) {
+            if (!localDiscoveryFuture.isCancelled()) {
+                localDiscoveryFuture.cancel(true);
+                localDiscoveryFuture = null;
```
Comment (jlaur) [inline bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java:85]:
```suggestion
                this.backgroundDiscoveryFuture = null;
```

File: bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java [hunk 3]
```diff
@@ -63,20 +67,23 @@ protected void startScan() {
...
-        super.stopBackgroundDiscovery();
-    }
 
-    private void cancelBackgroundDiscoveryFuture() {
-        if (backgroundDiscoveryFuture != null && !backgroundDiscoveryFuture.isDone()) {
-            backgroundDiscoveryFuture.cancel(true);
-            backgroundDiscoveryFuture = null;
+        ScheduledFuture<?> localDiscoveryFuture = backgroundDiscoveryFuture;
+        if (localDiscoveryFuture != null) {
+            if (!localDiscoveryFuture.isCancelled()) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java:83]: There is no need to check this.

File: bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java [hunk 4]
```diff
@@ -105,27 +112,35 @@ private void scan() {
...
-            ThingUID thingUID = new ThingUID(AirVisualNodeBindingConstants.THING_TYPE_AVNODE,
-                    nodeSerialNumber.toLowerCase());
-
-            try {
-                // Get the Node address by name
-                NbtAddress nodeNbtAddress = NbtAddress.getByName(serverName);
-                if (nodeNbtAddress == null) {
-                    // The Node address not found by some reason, skip it
-                    continue;
+            if (nodeSerialNumber != null) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java:115]: Consider reducing nesting:
```suggestion
            if (nodeSerialNumber == null) {
                logger.debug("Extracting the Node serial number failed");
                return;
            }
```

File: bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/handler/AirVisualNodeHandler.java [hunk 5]
```diff
@@ -173,14 +173,15 @@ public void dispose() {
     }
 
     private synchronized void stopPoll() {
-        if (pollFuture != null && !pollFuture.isCancelled()) {
-            pollFuture.cancel(false);
+        ScheduledFuture<?> localFuture = pollFuture;
+        if (localFuture != null && !localFuture.isCancelled()) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/handler/AirVisualNodeHandler.java:177]: There is no need to check if cancelled before cancelling it.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the refactoring. I have provided a few comments. You might want to test discovery again, or correct my comments. :-)

File: bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java:72]: No this was not meant to be the case, i accidently grabbed the 600 from the singleto at line 58 i guess. Thanks you saved me here :-)

File: bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java (hunk 4 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java:115]: I think it is just some structure i (wrongly) tought myself. I'll try to remember this for all PR's ill do in the future now ;-)

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**